src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.Log.cs

```diff
@@ -52,14 +57,30 @@ private static void ConvertToLocalTimes(object[] values)
 
             public bool Fail(string reason, string message, params object[] values)
             {
+                _stopwatch.Stop();
+
                 Minimal(message, values);
-                _telemetryService.PostProperty(TelemetryEventName.UpToDateCheckFail, TelemetryPropertyName.UpToDateCheckFailReason, reason);
+
+                _telemetryService.PostProperties(TelemetryEventName.UpToDateCheckFail, new[]
+                {
+                    (TelemetryPropertyName.UpToDateCheckFailReason, (object)reason),
+                    (TelemetryPropertyName.UpToDateCheckDurationMillis, _stopwatch.Elapsed.TotalMilliseconds),
```

Comment:
>UpToDateCheckDurationMillis [](start = 43, length = 27)

It's a little bit of a shame that the project system telemetry service doesn't support telemetry events with duration (i.e. a start and an end) because then this would just be recorded naturally by the telemetry event.